### ISSUE TITLE
Missing packages - Debian 11 (bullseye)

### ISSUE THREAD

**Issue by elderlabs** [eLvErDe/hwraid#129]
Looking to install packages on Debian 11. Seems only two exist in the group that I'm looking for.
`megaclisas-status` and `megacli` exist and install properly, while the following could not be found.

```
megaraid-status megactl megamgr dellmgr lsiutil
```

For the most part, I'm looking for `lsiutil` and `dellmgr`. Any ideas as to what's going on? Thanks.

**Comment by Jibun-no-Kage** [comment]
Same here... need to support PERC4 controllers and seems megaraid-status and megactl are gone from the repo for bullseye?

**Comment by eLvErDe** [comment]
I'm just missing some time :/

I'll try to handle this soon

**Comment by elderlabs** [comment]
Workaround seems to be installing from the `buster` repo. These packages seem to function normally.

**Comment by Jibun-no-Kage** [comment]
Yeah... I happen to have done just that, pulled the files from buster.  Was able to access a PERC4 controller via CLI, create a RAID 5 set, and initialize it.

**Comment by St0nefish** [comment]
Commenting to keep this alive. I was looking for sas2ircu to help locate a disk with errors in my ZFS array on Proxmox 7.2/Debian 11. I was also able to work around by setting release to Buster.

Thanks for providing these ports. 

For anyone in a similar boat, the sas2ircu utility works great to enable the locate LED on my Supermicro 846 chassis with SAS expander backplane using a LSI 9207-8i controller. *Much* better than randomly pulling disks and checking ZFS status until I get the right one. 

Edit: I started from the Debian Packages page here (https://hwraid.le-vert.net/wiki/DebianPackages) which does not actually list Buster as a supported release. Stretch worked as well (it appears to be the exact same package version) but FYI the page is out of date.

**Comment by telmnstr** [comment]
I ended up using the buster tw-cli package on bullseye and it works as far as showing status and such. But when I issue a "/c0 rescan" I get a segfault. Broadcom now owns the 3ware assets via Avago > LSI > AMCC > 3ware I guess but didn't see much on their site. From what I can find, the source for tw-cli isn't out there.

strace shows:
`open("/var/log/tw_mgmt.log", O_WRONLY|O_CREAT|O_APPEND, 0666) = 3`
`fstat(3, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0`
`mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fa761ccf000`
`fstat(3, {st_mode=S_IFREG|0644, st_size=0, ...}) = 0`
`lseek(3, 0, SEEK_SET)                   = 0`
`--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0xffffffffff600400} ---`
`+++ killed by SIGSEGV +++`
`Segmentation fault`

I have tried to manually create the /var/log/tw_mgmt.log file and adjust permissions but no go there.

Also prior to that part it seems to loop over and over this function:

`open("/dev/twa0", O_RDWR)               = 3`
`uname({sysname="Linux", nodename="debian", ...}) = 0`
`ioctl(3, _IOC(_IOC_NONE, 0x1, 0x8, 0), 0x1082c80) = 0`
`close(3)                         `

**Comment by DerBunteBall** [comment]
I can confirm what @telmnstr experiences. Nearly same here.

I'm getting segfaults at least for every "more complex" operation then showing infos. The log files gets created but stays empty. Owner is root and permissions are 644.

Tested the Buster package and tried version 9.5 and 10.2 (found on Hoster tool server - version numbers were the naming of them I think) everywhere the same. I'm actually don't know which of them created the log file.

Controller is a relatively old 9650SE-8LPML.

Driver Version: 2.26.02.014
Firmware Version: FE9X 4.10.00.027
BIOS Version: BE9X 4.08.00.004
Boot Loader Version: BL9X 3.08.00.001

Don't did it but I'm sure I would get the same results which @telmnstr got with strace.

Does someone know a tool/build which actually works on Debian Bullseye and Bookworm for 3Ware?

Best Regards

**Comment by DerBunteBall** [comment]
Use [this](https://ftp.fau.de/gentoo/distfiles/56/368_tw_cli_debian8_beta.tgz)

That's the newer build of tw_cli for Jessie. This works for me on Bullseye with the above mentioned controller. I have no segfaults until now also with complex operations.

I think this thing was posted earlier but the Push link is down.

Found this [here](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-block/tw_cli/tw_cli-10.2.2.1-r2.ebuild) in the Gentoo ebuild file for tw_cli.

Eventually this helps someone. Seems this build can work with newer kernels without problems. My kernel on Bullseye is 5.10.0-26-amd64

Best Regards
